### PR TITLE
improve authentication layer stability

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
This PR attempts to make our authentication layer (Dex + Keycloak-Gatekeeper) more stable by

* defining PDBs for both components
* allowing to specify the replica count for Keycloak deployments
* raising the default replica count to 2
* setting affinities to prevent two pods to be scheduled on the same node

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by xrstf** [comment]
/hold

Need to check if we can actually define meaningful affinities in this chart.

File: config/iap/values.yaml [hunk 1]
```diff
@@ -1,7 +1,12 @@
 iap:
+  # replicas per deployment; you can set this explicitly per deployment
+  # to override this
+  replicas: 1
+
   deployments:
     # alertmanager:
     #   name: alertmanager
+    #   replicas: 2 #
```
Comment (kron4eg) [inline config/iap/values.yaml:9]: Is it really needed to have individual replicas?

Comment (xrstf) [reply config/iap/values.yaml:9]: Yes.

Comment (kron4eg) [reply config/iap/values.yaml:9]: But I don’t see how, could you please explain a bit more.

Comment (xrstf) [reply config/iap/values.yaml:9]: There can be IAPs in front of services on preemtible nodes where one or more nodes die, whereas some other IAPs are running in front of apps that are on stable nodes, where only a single replica is enough. That was my motivation for adding this, given how quick and easy it was to do.

**Review by kron4eg — APPROVED** [review]
/lgtm 
/approve no-issue

**Pull request merged**